src/App.tsx

```diff
@@ -14,37 +16,105 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const REVERSE = false;
+
+enum SortType {
+  RESET_VALUE = '',
+  SORT_BY_NAME = 'name',
+  SORT_BY_LENGTH = 'length',
+}
+
+interface SortCriteria {
+  sortBy: SortType;
+  isReversed: boolean;
+}
+
+function sortGoods(
+  goods: string[],
+  { sortBy: criteria, isReversed }: SortCriteria,
+) {
+  let sortedGoods;
+
+  switch (criteria) {
+    case SortType.SORT_BY_NAME:
+      sortedGoods = goods.toSorted((a, b) => a.localeCompare(b));
+      break;
+    case SortType.SORT_BY_LENGTH:
+      sortedGoods = goods.toSorted((a, b) => a.length - b.length);
```

Comment:
Similar to the previous comment, the correct method name is `sort`, not `toSorted`.